package.json

```
@@ -287,14 +287,14 @@
 		"*.scss": [
 			"wp-scripts lint-style"
 		],
-		"*.js": [
+		"*.{js,ts}": [
```

I think we could add `.tsx` as well. TS that includes JSX _requires_ the `.tsx` extension. I'd expect this to work seamlessly when and if a `.tsx` file is added and it comes at little or no cost now:
```suggestion
		"*.{js,ts,tsx}": [
```

---

It looks like we need to ignore `.d.ts` - we're getting bad `eslint` errors for this `Unexpected var, use let or const instead. eslint(no-var)`:

https://github.com/WordPress/gutenberg/blob/3ec7aa5356dc947636d0e7426e3453ad5c654eab/typings/gutenberg-env/index.d.ts#L7

---

What would be the case when defining types to use `tsx`? I'm mostly curious because this decision doesn't prevent someone from creating such a file so in theory, it's better to have it rather than not 😄

---

Added `*.d.ts` to `.eslintignore`.

---

> What would be the case when defining types to use tsx?

There's no reason to use `.tsx` extension unless the file will include JSX syntax. When strictly describing types, I don't think there's any reason to use `.tsx`. However, there's nothing in the system that prevents such a file from being added. It seems preferable to cover the possibility (maybe redundantly) instead of leaving it open.

---

I'm fine with your making the decision 👍